cmd/nitro/nitro.go

```diff
@@ -454,15 +454,20 @@ func mainImpl() int {
 		return 1
 	}
 
-	consensusDB, err := stack.OpenDatabaseWithOptions("arbitrumdata", node.DatabaseOptions{MetricsNamespace: "arbitrumdata/", PebbleExtraOptions: nodeConfig.Persistent.Pebble.ExtraOptions("arbitrumdata"), NoFreezer: true})
-	deferFuncs = append(deferFuncs, func() { closeDb(consensusDB, "consensusDB") })
-	if err != nil {
-		log.Error("failed to open database", "err", err)
-		log.Error("database is corrupt; delete it and try again", "database-directory", stack.InstanceDir())
-		return 1
-	}
+	// TODO: Have something similar to the Execution side where we first call
+	// openExistingExecutionDB() and if that fails we call downloadDB(). In
+	// this case we would be calling openExistingConsensusDB() and if that
+	// fails we would either call downloadDB() or downloadConsensusDB() which
+	// is yet to be implemented:
+	// if consensus-node=true {
+	//		arbDb = openExistingConsensusDB(...)
+	//		if arbDb == nil {
+	//			downloadConsensusDB(...)
+	//			arbDb = openDownloadedConsensusDB(...)
+	//		}
+	// }
+	consensusDB, err := openDownloadedConsensusDB(stack, nodeConfig, &deferFuncs)
 	if err := dbutil.UnfinishedConversionCheck(consensusDB); err != nil {
```

Comment:
This check is being done twice, here and inside openDownloadedConsensusDB.
Once is enough.